crates/ruff_linter/src/fix/codemods.rs

```diff
@@ -167,39 +170,46 @@ pub(crate) fn retain_imports(
     Ok(tree.codegen_stylist(stylist))
 }
 
-fn collect_segments<'a>(expr: &'a Expression, parts: &mut SmallVec<[&'a str; 8]>) {
-    match expr {
-        Expression::Call(expr) => {
-            collect_segments(&expr.func, parts);
-        }
-        Expression::Attribute(expr) => {
-            collect_segments(&expr.value, parts);
-            parts.push(expr.attr.value);
-        }
-        Expression::Name(expr) => {
-            parts.push(expr.value);
+fn qualified_name_from_name_or_attribute(module: &NameOrAttribute) -> String {
+    fn collect_segments<'a>(expr: &'a Expression, parts: &mut SmallVec<[&'a str; 8]>) {
+        match expr {
+            Expression::Call(expr) => {
+                collect_segments(&expr.func, parts);
+            }
+            Expression::Attribute(expr) => {
+                collect_segments(&expr.value, parts);
+                parts.push(expr.attr.value);
+            }
+            Expression::Name(expr) => {
+                parts.push(expr.value);
+            }
+            _ => {}
         }
-        _ => {}
     }
-}
 
-fn unqualified_name_from_expression<'a>(expr: &'a Expression<'a>) -> Option<UnqualifiedName<'a>> {
-    let mut segments = smallvec![];
-    collect_segments(expr, &mut segments);
-    if segments.is_empty() {
-        None
-    } else {
-        Some(segments.into_iter().collect())
+    fn unqualified_name_from_expression<'a>(
+        expr: &'a Expression<'a>,
+    ) -> Option<UnqualifiedName<'a>> {
+        let mut segments = smallvec![];
+        collect_segments(expr, &mut segments);
+        if segments.is_empty() {
+            None
+        } else {
+            Some(segments.into_iter().collect())
+        }
     }
-}
 
```

Comment:
What's the motivation for nesting these functions? Seems a fairly unrelated change and it took me quiet a bit to figure out **what** changed.

Reply:
Sorry again for taking this out of review mode before I had written a PR summary. The motivation here is that the `UnqualifiedName` produced by `unqualified_name_from_expression` should also strictly-speaking be NFKC-normalized before it's safe to be used elsewhere and potentially compared to other `UnqualifiedName`s; the same applies for the segments collected by `collect_segments()`. Currently we only use these functions from the `qualified_name_from_name_or_attribute()`, so I think it's okay (and _probably_ more efficient? it allocates...) to just do a single NKFC normalization pass at the end of `qualified_name_from_name_or_attribute()`, _as long as_ we enforce that `unqualified_name_from_expression()` and `collect_segments()` can't be called from other functions.

Does that make sense? I suppose I should probably add a comment.

Reply:
Kind of. I'm not that concerned because the functions are private anyway. But I don't feel strongly about it.

Reply:
Yeah, I was mostly concerned that other functions in this file might start using them.